File: DataBindingExample/app/src/main/java/com/hasib/databindingexample/MainActivity.java
package com.hasib.databindingexample;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import com.hasib.databindingexample.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        SimpleViewModel viewModel = new ViewModelProvider(this, new ViewModelProvider.NewInstanceFactory()).get(SimpleViewModel.class);

        binding.setLifecycleOwner(this);
        binding.setViewmodel(viewModel);
    }
}

File: _Kotlin-Camouflage/TestApplication/app/src/main/java/com/hasib/testapplication/MainActivity.kt
package com.hasib.testapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.lifecycleScope
import com.hasib.testapplication.databinding.ActivityMainBinding
import kotlinx.coroutines.*
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.flow.flow
import kotlin.coroutines.coroutineContext

class MainActivity : AppCompatActivity() {
    
    private val TAG = "MainActivity"

    @ObsoleteCoroutinesApi
    private val scope = CoroutineScope(newFixedThreadPoolContext(2, "scope"))
    
    @ObsoleteCoroutinesApi
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)

        lifecycleScope.launch {
            withContext(Dispatchers.Default) {
                flow<Int> {
                    for (i in 1..5) {
                        delay(1000)
                        emit(i)
                    }
                }.collect { binding.value = it }
            }
            Log.d(TAG, "onCreate: After flow")
        }
        Log.d(TAG, "onCreate: After Blocking")
    }
}